stdlib/public/core/Span/Span.swift

```diff
@@ -0,0 +1,827 @@
+  }
+
+  /// Unsafely creates a `Span` over initialized memory.
+  ///
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
+  ///
+  /// `buffer` must be correctly aligned for accessing
+  /// an element of type `Element`, and must contain a number of bytes
+  /// that is an exact multiple of `Element`'s stride.
+  ///
+  /// - Parameters:
+  ///   - buffer: a buffer to initialized elements.
+  ///   - type: the type to use when interpreting the bytes in memory.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
+  public init(
+    _unsafeBytes buffer: Slice<UnsafeMutableRawBufferPointer>
+  ) {
+    self.init(_unsafeBytes: UnsafeRawBufferPointer(rebasing: buffer))
+  }
+}
+
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 9999, *)
+extension Span where Element: Equatable {
+
+  /// Returns a Boolean value indicating whether this and another span
+  /// contain equal elements in the same order.
+  ///
+  /// - Parameters:
+  ///   - other: A span to compare to this one.
+  /// - Returns: `true` if this sequence and `other` contain equivalent items,
+  ///   using `areEquivalent` as the equivalence test; otherwise, `false.`
+  ///
+  /// - Complexity: O(*m*), where *m* is the lesser of the length of the
+  ///   sequence and the length of `other`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _elementsEqual(_ other: Self) -> Bool {
+    guard count == other.count else { return false }
+    if count == 0 { return true }
+
+    //FIXME: This could be short-cut
+    //       with a layout constraint where stride equals size,
+    //       as long as there is at most 1 unused bit pattern.
+    // if Element is BitwiseEquatable {
+    // return _swift_stdlib_memcmp(lhs.baseAddress, rhs.baseAddress, count) == 0
+    // }
+    for o in 0..<count {
+      if self[unchecked: o] != other[unchecked: o] { return false }
+    }
+    return true
+  }
+
+  /// Returns a Boolean value indicating whether this span and a Collection
+  /// contain equal elements in the same order.
+  ///
+  /// - Parameters:
+  ///   - other: A Collection to compare to this span.
+  /// - Returns: `true` if this sequence and `other` contain equivalent items,
+  ///   using `areEquivalent` as the equivalence test; otherwise, `false.`
+  ///
+  /// - Complexity: O(*m*), where *m* is the lesser of the length of the
+  ///   sequence and the length of `other`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _elementsEqual(_ other: some Collection<Element>) -> Bool {
+    guard count == other.count else { return false }
+    if count == 0 { return true }
+
+    return _elementsEqual(AnySequence(other))
```

Comment:
This should probably try to use withContiguousStorageIfAvailable to escape generics world if possible and fall back to AnySequence only if everything else failed.

Reply:
Done